modyn/supervisor/internal/grpc_handler.py

```diff
@@ -434,3 +462,169 @@ def seed_selector(self, seed: int) -> None:
         ).success
 
         assert success, "Something went wrong while seeding the selector"
+
+    def start_evaluation(self, trained_model_id: int, pipeline_config: dict) -> dict[int, EvaluationStatusTracker]:
+        if not self.connected_to_evaluator:
+            raise ConnectionError("Tried to start evaluation at evaluator, but there is no gRPC connection.")
+
+        model_id = pipeline_config["model"]["id"]
+        if "config" in pipeline_config["model"]:
+            model_config = json.dumps(pipeline_config["model"]["config"])
+        else:
+            model_config = "{}"
+
+        device = pipeline_config["evaluation"]["device"]
+        amp = pipeline_config["evaluation"]["amp"] if "amp" in pipeline_config["evaluation"] else False
+
+        evaluations: dict[int, EvaluationStatusTracker] = {}
+
+        for dataset in pipeline_config["evaluation"]["datasets"]:
+            dataset_id = dataset["dataset_id"]
```

Comment:
could you maybe put the body of that loop into an extra function? sorry, I think i just did not manage to read it well and hence made the wrong claim that there is an indentation issue. if we have one function to call per dataset it should be clearer

Reply:
Sure, done!